packages/interactivity/src/directives.js

```diff
@@ -8,7 +8,7 @@ import { deepSignal, peek } from 'deepsignal';
  * Internal dependencies
  */
 import { createPortal } from './portals';
-import { useWatch, useInit } from './utils';
+import { useWatch, useInit, useEffect } from './utils';
```

Comment:
If you're going to use `useEffect` from `@wordpress/interactivity`, you can use `useInit` instead.

@DAreRodz: what do we do? Do we keep using `useWatch` and `useInit` in the directives? Or do we switch back to hooks that don't have an unnecessary call to `withScope`? To me, it feels weird to have that double set of the scope.

Reply:
If there are cases in which we need `useWatch` and `useInit`. It may be worth the cost of that `withScope` and simplify all the directives with one solution.

Reply:
Yeah, I'm still thinking about what approach is better. I would prefer the same hooks for directives and store callbacks, but I agree that double-setting the same scope feels weird. 🤔

Let's consider the DevX.
- If we stay with the same hooks and `evaluate`, everything is transparent for the developer (although internally _weird things_ happen).
- If we have multiple hook versions, we would have to clarify which one is used in which case. And, if the API is meant to be publicly exposed from `@wordpress/interactivity`, we should use different names. E.g. `useEffect` and `useScopedEffect`. And maybe you could have more complex cases where you need to use a scoped hook and call `evaluate()` inside, leading to the same weirdness as the first option (I haven't thought of any, but who knows).
- We can also get rid of the scoped versions and expose a `withScope()` function (maybe even for `evaluate`) so devs explicitly decide when the scope is available.
   ```js
   withScope( evaluate )( entry );
   useEffect( withScope( () => {
     // Do something.
   } );
   ```

Any other ideas? 😄

Reply:
I'm just talking about our `directives.js` file.

Replace this:

```js
import { useInit } from "..";

// ...

// data-wp-init--[name]
directive("init", ({ directives: { init }, evaluate }) => {
  init.forEach((entry) => {
    useInit(() => evaluate(entry));
  });
});
```

with this:

```js
import { useEffect } from "preact/hooks";

// ...

// data-wp-init--[name]
directive("init", ({ directives: { init }, evaluate }) => {
  init.forEach((entry) => {
    useEffect(() => evaluate(entry), []);
  });
});
```

Exporting our own `useInit`, `useWatch`, `useEffect`, `useMemo`… with integrated `withScope` is fine, we should keep doing that, because people using them in `data-wp-run` won't use them with `evaluate`.